docs/design/2020-10-22-local-transaction.md

```diff
@@ -0,0 +1,129 @@
+# Proposal: Support the Local/Global Transaction in Cross-DC Deployment
+
+- Author(s):     [Haizhi/JmPotato](https://github.com/JmPotato)
+- Last updated:  2020-10-22
+- Discussion at: https://github.com/pingcap/tidb/issues/20561
+
+## Abstract
+
+Make TiDB support the Local and Global transactions.
+
+## Background
+
+As a part of [tidb/issues/18273](https://github.com/pingcap/tidb/issues/18273), now PD has the ability to allocate the Local/Global TSO to support the corresponding Local/Global linear consistency (for more details, you can look at [pd/issues/2759](https://github.com/tikv/pd/issues/2759)). Based on this, TiDB could start two kinds of transactions: local and global.
+
+* Local transaction: only allow to read & write the local data which come from the same region of a local TSO
+* Global transaction: allow to read & write any data in a cluster with a Global TSO
+
+By supporting these two kinds of transactions, we can reduce the TSO fetching latency caused by the cross-region deployment to improve the transaction performance. So we need TiDB to have a mechanism to support starting these two kinds of transactions.
+
+## Proposal
+
+### Configuration
+
+Since the transaction process is managed on the TiDB side, when initiating a transaction, we need to know the topology information of the current TiDB instance as a PD Client so that we can know which location of the PD to request. Like PD, TiDB needs to introduce a field such as ***dc-location*** in the label configuration. This label is string type, the default value is empty, and used to set which DC the TiDB node belongs to, so that it can distinguish local/global transaction needs when performing TSO services.
+
+And because this label may be used by Auto Scaling or other components, we need to add a switch to control the Local/Global transaction enable/disable in the configuration which will be named as enable-local-transaction, bool type and the default value is false.
```

Comment:
```suggestion
As the label configuration could be modified by other purpose, we will also provide a switch to enable/disable the Local/Global transaction ability in the configuration. It will be named as enable-local-transaction, bool type and the default value is false.
```